lib/storage/client.js

```diff
@@ -1182,6 +1182,85 @@ class Client extends CryptoConsumer {
     })
     return listGroups
   }
-}
 
+  /**
+   * addGroupMembers adds members to the TozStore Group given.
+   *
+   * @param {string} groupId  UUID assigned by TozStore, used to identify a group.
+   * @param {GroupMember[]} groupMembers wraps together the ClientID, Capability and Membership Key for each client being added to the group
+   *
+   *
+   * @returns {GroupMember} A response from TozStore; the group members who have been added.
+   *
+   */
+  async addGroupMembers(groupId, groupMembers = []) {
+    // Get Group Membership key for current client
+    let membershipResponse = await this.authenticator.tsv1Fetch(
+      `${this.config.apiUrl}/v2/storage/groups/${groupId}/membership_key`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let groupMembershipJson = await validateResponseAsJSON(membershipResponse)
+    // Get All members Public Keys to create their membership keys
+    let publicKeysResponse = await this.authenticator.tokenFetch(
+      `${this.config.apiUrl}/v1/client/${groupMembers.clientID}/public`,
+      {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      }
+    )
+    let publicKeysJSON = await validateResponseAsJSON(publicKeysResponse)
+    // GET THE PUBLIC KEY FOR THE AUTHORIZER
+    let publicKeysAuthResp = await this.authenticator.tokenFetch(
+      `${this.config.apiUrl}/v1/client/${groupMembershipJson.authorizer_id}/public`,
```

Comment:
Who is the authorizer in this scenario and why do we need their information here, isn't the caller of this method the one adding new group members and thus the authorizer for these clients?

Reply:
we need to know who wrote the membership key for the person adding these people. Since we need their public key in order to unwrap the membership key